src/lib/user/user.ts

```diff
@@ -10,17 +13,80 @@ export abstract class User implements DefaultUser {
 		)!;
 	}
 
-	public static changeUserClass(userClass: new (ref: Player | number) => User): void {
+	async load(): Promise<unknown> {
+		return ""; /** What it is "loading" itself? And from "where"? */
+	}
+
+	async unload(): Promise<void> {}
+}
+
+class OfflineUser extends User implements OfflineUserDeclaration {
+	release(): void {
+		/** Somehow, release the offline user if no longer needed */
+	}
+}
+
+export namespace Users {
+	const usersMap: Map<Player, DefaultUserDeclaration> = new Map();
+
+	let TINA_USER_CLASS = User as never as new (ref: Player | number) => DefaultUserDeclaration & unknown;
+
+	export async function add(player: Player) {
+		const user = new TINA_USER_CLASS(player);
+		await user
+			.load()
+			.catch((e) =>
+				logger.warn(`[User]: Player's data couldn't be loaded correctly. More information: ${e}`),
+			); /** User load? */
+
+		usersMap.set(player, user);
+	}
+
+	export async function remove(player: Player) {
+		const user = usersMap.get(player);
+		if (user !== undefined) {
+			await user
+				.unload()
+				.catch((e) =>
+					logger.warn(`[User]: Player's data couldn't be unloade correctly. More information: ${e}`),
+				); /** User unload? */
+		}
+
+		usersMap.delete(player);
+	}
```

Comment:
We would also fire the events like `Tina.when("user:added" or "user:removed")` around here.